src/migrate/migrate_to_asset_definition_v2.rs

```diff
@@ -0,0 +1,284 @@
+use crate::core::msg::MigrationOptions;
+use crate::core::state::{asset_definitions, insert_asset_definition_v2};
+use crate::core::types::asset_definition::AssetDefinitionV2;
+use crate::migrate::migrate_contract::migrate_contract;
+use crate::util::aliases::{DepsMutC, EntryPointResponse};
+use crate::util::traits::ResultExtensions;
+use cosmwasm_std::Order;
+
+pub fn migrate_to_asset_definition_v2(
+    deps: DepsMutC,
+    options: Option<MigrationOptions>,
+) -> EntryPointResponse {
+    let asset_definitions = asset_definitions()
+        .range_raw(deps.storage, None, None, Order::Descending)
+        .map(|item| item.unwrap().1.to_v2())
+        .collect::<Vec<AssetDefinitionV2>>();
+    let migrated_def_count = asset_definitions.len();
+    for definition in asset_definitions.into_iter() {
+        insert_asset_definition_v2(deps.storage, &definition)?;
+    }
+    // Pass through the existing migration code to ensure version upgrades also occur
```

Comment:
Should we also go through and delete the old asset definitions here? I'm not sure if there is any downside to leaving them hanging around necessarily, but...

Reply:
Good call - I added this and tested that it works because we've never done that before lol.  Should work just fine